src/Core/Application/Common/Interfaces/IRepositoryAsync.cs

```diff
@@ -1,316 +1,338 @@
+    Task<TProjectedType?> GetAsync<T, TProjectedType>(Expression<Func<T, TProjectedType>> selectExpression, BaseSpecification<T>? specification = null, bool asNoTracking = true, CancellationToken cancellationToken = default)
+    where T : BaseEntity;
+
+    /// <summary>
+    /// Get the count of <typeparamref name="T"/> based on the criteria specified in <paramref name="condition"/>.
+    /// </summary>
+    /// <typeparam name="T">The type of the entity.</typeparam>
+    /// <param name="condition">The condition on which the returned count of <typeparamref name="T"/> will be filtered.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to observe while waiting for the task to complete.</param>
+    /// <returns>Returns <see cref="Task"/> of <see cref="int"/>.</returns>
+    Task<int> GetCountAsync<T>(Expression<Func<T, bool>>? condition = null, CancellationToken cancellationToken = default)
+    where T : BaseEntity;
+
+    /// <summary>
+    /// Determine whether <typeparamref name="T"/> exists based on the criteria specified in <paramref name="condition"/>.
+    /// </summary>
+    /// <typeparam name="T">The type of the entity.</typeparam>
+    /// <param name="condition">The condition on which the existence of <typeparamref name="T"/> will be determined.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to observe while waiting for the task to complete.</param>
+    /// <returns>Returns <see cref="Task"/> of <see cref="bool"/>.</returns>
+    Task<bool> ExistsAsync<T>(Expression<Func<T, bool>> condition, CancellationToken cancellationToken = default)
+    where T : BaseEntity;
+
+    /// <summary>
+    /// Add a single <typeparamref name="T"/> to the EF Core context to be persisted in the database.
+    /// </summary>
+    /// <typeparam name="T">The type of the entity.</typeparam>
+    /// <typeparam name="TKey">The type of the entity id.</typeparam>
+    /// <param name="entity">The entity to be inserted.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to observe while waiting for the task to complete.</param>
+    /// <returns>Returns <see cref="Task"/> of <see cref="TKey"/>.</returns>
+    Task<TKey> CreateAsync<T, TKey>(T entity, CancellationToken cancellationToken = default)
+    where T : BaseEntityWith<TKey>;
+
+    /// <summary>
+    /// Add a single <typeparamref name="T"/> to the EF Core context to be persisted in the database.
+    /// </summary>
+    /// <typeparam name="T">The type of the entity with id of type int.</typeparam>
+    /// <param name="entity">The entity to be inserted.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to observe while waiting for the task to complete.</param>
+    /// <returns>Returns <see cref="Task"/> of <see cref="int"/>.</returns>
+    Task<int> CreateAsync<T>(T entity, CancellationToken cancellationToken = default)
+    where T : BaseEntityWith<int>;

```

Comment:
I see you're using int here as the default type... while the default is now Guid...

We might be able to do something like 
```c#
    global using DefaultIdType = System.Guid;
```
and the use DefaultIdType everywhere where we now use Guid (including here where you use int)... so it can easily be changed in one place.

Reply:
That's a good idea, but maybe in next PR
unfortunately I don't have any more time to maintain this PR

Reply:
That's unfortunate indeed... We can keep it as a reference for when it gets implemented.